### PR TITLE
MPR/ASR, avoid calculations when not needed, fix a bug

### PR DESCRIPTION
**Pull request by sravfeyn**
@Rohit25negi 
@emord

### CODE REVIEW

**Review by snopoke — APPROVED** [review]
Don't have a lot of context but code looks fine

File: custom/icds_reports/mpr_sqldata.py [hunk 1]
```diff
@@ -1633,15 +1633,16 @@ class MPRImmunizationCoverage(ICDSMixin, MPRData):
...
-        data = self.custom_data(selected_location=self.selected_location, domain=self.config['domain'])
-        children_completing = data.get('open_child_count', 0)
-        vaccination = data.get('open_child_1yr_immun_complete', 1)
-        immunization = "%.1f%%" % ((children_completing / float(vaccination or 1)) * 100)
-        return [
-            ['(I)', 'Children Completing 12 months during the month:', children_completing],
-            ['(II)', 'Of this, number of children who have received all vaccinations:', vaccination],
-            ['(III)', 'Completed timely immunization coverage (%):', immunization]
-        ]
+        if self.config['location_id']:
```
Comment (snopoke) [inline custom/icds_reports/mpr_sqldata.py]: if this returns `None` is that handled by the caller?

File: custom/icds_reports/reports/__init__.py [hunk 2]
```diff
@@ -104,6 +104,8 @@ def report_context(self):
             }
 
     def get_report_context(self, data_provider):
+        if not self.filter_set:
```
Comment (snopoke) [inline custom/icds_reports/reports/__init__.py]: when would this happen?

Comment (sravfeyn) [reply custom/icds_reports/reports/__init__.py]: When the report page is loaded (without filters set).

File: custom/icds_reports/mpr_sqldata.py (hunk 1 above)
Comment (sravfeyn) [reply custom/icds_reports/mpr_sqldata.py]: Yes. This is how the rows method on all the rest of subreports is implemented, it's not in a great shape, but I am avoiding too much refactors on MPR/ASR

**Pull request merged**